### ISSUE TITLE
The factory definition in not callable

### ISSUE THREAD

**Issue by userlond** [PHP-DI/PHP-DI#342]
Hi. I try to configure container:

```php
// app/di.php
<?
use function DI\factory;

return [
    Doctrine\ORM\EntityManager::class => factory(App\Factory\DoctrineFactory::class)
];
```
```php
<?
// app/bootstrap.php
use DI\ContainerBuilder;

require 'vendor/autoload.php';

$containerBuilder = new ContainerBuilder;
$containerBuilder->addDefinitions('app/di.php');
$container = $containerBuilder->build();
return $container;
```
```php
// src/App/Factory/DoctrineFactory.php, autoloaded with psr-0
<?
namespace App\Factory;

use Doctrine\ORM\EntityManager;
use Interop\Container\ContainerInterface;

class DoctrineFactory
{
    public function __invoke(ContainerInterface $c)
    {
        // create EntityManager here

        return EntityManager::create($conn, $config);
    }
}
```
When I get doctrine via `$container->get(Doctrine\ORM\EntityManager::class)` I got an error:

Php Fatal error: Uncaught exception ...  The factory definition "Doctrine\ORM\EntityManager" is not callable.

When I replaced __invoke with static create method in my factory and modify app/di.php, everything works:

```php
// app/di.php
<?
use function DI\factory;

return [
    Doctrine\ORM\EntityManager::class => factory('App\Factory\DoctrineFactory::create')
];
```
Is it a bug or my mistake? Thanks for help.

**Comment by mnapoli** [comment]
Hi, thanks for the report. That's indeed not normal, was it with the latest version?

**Comment by mnapoli** [comment]
Oh wait sorry! This feature is added in PHP-DI 5.2 which is not released yet (#321).

**Comment by userlond** [comment]
Thanks for reply and avewome product.

I switched from "php-di/php-di": "^5.1" to "php-di/php-di": "dev-master" and now it works!
